dags/maxtext_pathways/configs/utils.py

```
@@ -188,3 +257,228 @@ def check_gcp_logs_exist(
+      if pattern in clean_line:
+        logging.info("Target log pattern '%s' matched!", pattern)
+
+        if "Z " in clean_line:
+          return clean_line.split("Z ")[0] + "Z"
+
+        return datetime.now(timezone.utc).strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    # Process ended naturally (container finished) without finding the string
+    raise AirflowException(
+        f"Log stream ended, but pattern '{pattern}' was not found."
+    )
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)